templates/vsphere/cluster-template.yaml

```diff
@@ -0,0 +1,1457 @@
+kind: Cluster
+metadata:
+  labels:
+    cluster.x-k8s.io/cluster-name: ${CLUSTER_NAME}
+  name: ${CLUSTER_NAME}
+  namespace: default
+spec:
+  clusterNetwork:
+    pods:
+      cidrBlocks:
+        - 192.168.0.0/16
+  controlPlaneRef:
+    apiVersion: controlplane.cluster.x-k8s.io/v1beta2
+    kind: CK8sControlPlane
+    name: ${CLUSTER_NAME}
+  infrastructureRef:
+    apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+    kind: VSphereCluster
+    name: ${CLUSTER_NAME}
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: VSphereCluster
+metadata:
+  name: ${CLUSTER_NAME}
+  namespace: default
+spec:
+  controlPlaneEndpoint:
+    host: ${CONTROL_PLANE_ENDPOINT_IP}
+    port: 6443
+  identityRef:
+    kind: Secret
+    name: ${CLUSTER_NAME}
+  server: ${VSPHERE_SERVER}
+  thumbprint: ${VSPHERE_TLS_THUMBPRINT}
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: VSphereMachineTemplate
+metadata:
+  name: ${CLUSTER_NAME}
+  namespace: default
+spec:
+  template:
+    spec:
+      cloneMode: linkedClone
+      datacenter: ${VSPHERE_DATACENTER}
+      datastore: ${VSPHERE_DATASTORE}
+      diskGiB: 25
+      folder: ${VSPHERE_FOLDER}
+      memoryMiB: 8192
+      network:
+        devices:
+          - dhcp4: true
+            networkName: ${VSPHERE_NETWORK}
+      numCPUs: 2
+      os: Linux
+      powerOffMode: trySoft
+      resourcePool: '${VSPHERE_RESOURCE_POOL}'
+      server: ${VSPHERE_SERVER}
+      storagePolicyName: ""
+      template: ${VSPHERE_TEMPLATE}
+      thumbprint: ${VSPHERE_TLS_THUMBPRINT}
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
+kind: VSphereMachineTemplate
+metadata:
+  name: ${CLUSTER_NAME}-worker
+  namespace: default
+spec:
+  template:
+    spec:
+      cloneMode: linkedClone
+      datacenter: ${VSPHERE_DATACENTER}
+      datastore: ${VSPHERE_DATASTORE}
+      diskGiB: 25
+      folder: ${VSPHERE_FOLDER}
+      memoryMiB: 8192
+      network:
+        devices:
+          - dhcp4: true
+            networkName: ${VSPHERE_NETWORK}
+      numCPUs: 2
+      os: Linux
+      powerOffMode: trySoft
+      resourcePool: '${VSPHERE_RESOURCE_POOL}'
+      server: ${VSPHERE_SERVER}
+      storagePolicyName: ""
+      template: ${VSPHERE_TEMPLATE}
+      thumbprint: ${VSPHERE_TLS_THUMBPRINT}
+---
+apiVersion: controlplane.cluster.x-k8s.io/v1beta2
+kind: CK8sControlPlane
+metadata:
+  name: ${CLUSTER_NAME}
+  namespace: default
+spec:
+  spec:
+    controlPlane:
+      cloudProvider: external
+    files:
+      - content: |
+          [Service]
+          Environment="HTTPS_PROXY=http://squid.internal:3128"
+          Environment="HTTP_PROXY=http://squid.internal:3128"
+          Environment="https_proxy=http://squid.internal:3128"
+          Environment="http_proxy=http://squid.internal:3128"
+          Environment="NO_PROXY=10.0.0.0/8,192.168.0.0/16,127.0.0.1,localhost,0.0.0.0"
+          Environment="no_proxy=10.0.0.0/8,192.168.0.0/16,127.0.0.1,localhost,0.0.0.0"
```

Comment:
two notes here:

- best move these to a secret (e.g. `${CLUSTER_NAME}-resources`) and then use `contentFrom` here to keep this section clean and manageable
- for the proxy configuration specifically, we cannot hardcode it into the template (having a vsphere environment is orthogonal to requiring a proxy). maybe you could add a ${VSPHERE_PROXY_DISABLED} prefix on all lines here, then add it to the list of vars as empty (to enable) or `# ` to disable.
-